backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -794,10 +794,28 @@ func (a *applicationEventLoopRunner_Action) applicationEventRunner_handleUpdateD
 	if applicationState.SyncError != "" {
 		condition.NewConditionManager().SetCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError, managedgitopsv1alpha1.GitOpsConditionStatusTrue, managedgitopsv1alpha1.GitopsDeploymentReasonSyncError, applicationState.SyncError)
 	} else {
-		// Update syncError Condition as false if applicationState.SyncError field in database is empty
-		reason := managedgitopsv1alpha1.GitopsDeploymentReasonSyncError + "Resolved"
-		if cond, _ := condition.NewConditionManager().FindCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError); cond.Reason != reason {
-			condition.NewConditionManager().SetCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError, managedgitopsv1alpha1.GitOpsConditionStatusFalse, reason, "")
+		// Update syncError Condition as false if applicationState.SyncError field in database is empty by checking if the condition field is empty or not
+		if condition.NewConditionManager().HasCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError) {
```

Comment:
If we are creating GitopsDeployment CR for the first time with no error , then it is going to add the condition without any error which is present inside if condition i.e, It will mark it as resolved without even getting the error so I am checking if the condition is present or no, if condition is present for eg, syncError, then we can update this error as resolved